sycl/doc/design/HostPipes.md

```diff
@@ -0,0 +1,193 @@
+document. This attribute is also used for other classes that have properties,
+so it is not specific to the `pipe` class. 
+
+The address of `static const __pipeType` member `__pipe` will be used to identify the pipe
+in host code, and provide one half of the host-to-device mapping of the pipe 
+(see the section on __New content in the integration header and footer__ below).
+
+[3]: <CompileTimeProperties.md>
+
+### Changes to the DPC++ front-end
+
+There are several changes to the device compiler front-end:
+
+* The front-end adds a new LLVM IR attribute `sycl-unique-id` to the definition
+  of each `pipe` variable, which provides a unique string identifier
+  for each.
+
+* The front-end generates new content in both the integration header and the
+  integration footer, which is described in more detail below.
+
+#### New content in the integration header and footer
+
+New content in the integration header and footer provides a mapping from the
+host address of each pipe variable to the unique string for that
+variable. To illustrate, consider a translation unit that defines two
+`pipe` classes:
+
+```
+#include <sycl/sycl.hpp>
+
+class some_pipe;
+namespace inner {
+  class some_other_pipe;
+} // namespace inner
+...
+pipe<class some_pipe, ...>::write(...); // a usage of pipe<class some_pipe, ...>
+...
+pipe<class some_other_pipe, ...>::read(...); // a usage of pipe<class some_other_pipe, ...> 
+...
+
+```
+
+The corresponding integration header defines a namespace scope variable of type
+`__sycl_host_pipe_registration` (referred to below as the __host pipe registrar__)
+whose sole purpose is to run its constructor before the application's main() function:
+
+```
+namespace sycl::detail {
+namespace {
+
+class __sycl_host_pipe_registration {
+ public:
+  __sycl_host_pipe_registration() noexcept;
+};
+__sycl_host_pipe_registration __sycl_host_pipe_registrar;
+
+} // namespace (unnamed)
+} // namespace sycl::detail
+```
+
+The integration footer contains the definition of the constructor, which calls
+a function in the DPC++ runtime with the following information for each host
+pipe that is used in the translation unit:
+
+* The (host) address of the static member variable `__pipe`.
+* The variable's string from the `sycl-unique-id` attribute.
+
+```
+namespace sycl::detail {
+namespace {
+
+__sycl_host_pipe_registration::__sycl_host_pipe_registration() noexcept {
+  host_pipe_map::add(&pipe<some_pipe, ...>::__pipe,
+    /* same string returned from __builtin_sycl_unique_pipe_id(pipe<some_pipe, ...>::__pipe) */);
+  host_pipe_map::add(&inner::pipe<some_other_pipe>::__pipe,
+    /* same string returned from __builtin_sycl_unique_pipe_id(pipe<some_other_pipe, ...>::__pipe) */);
+}
+
+} // namespace (unnamed)
+} // namespace sycl::detail
+```
+
+Further details on adherence to C++ rules for unconstructed objects can be found
+in the [device_global][3] design.
+
+[3]: <DeviceGlobal.md>
```

Comment:
Reference [3] is defined at line 83. This is detected by pre-commit.
Please, make sure that pre-commit checks passes.

Reply:
Fixed.

Reply:
Thanks!
Sorry for the delay.
Unfortunately, I don't see pre-commit results for this PR. Could you sync your branch with the tip of the `sycl` branch to trigger a new pre-commit validation in CI system, please?